src/reactor.rs

```diff
@@ -458,43 +509,71 @@ mod sys {
             fcntl(fd, FcntlArg::F_SETFD(FdFlag::FD_CLOEXEC)).map_err(io_err)?;
             Ok(Reactor(fd))
         }
-        pub fn register(&self, fd: RawFd, key: usize) -> io::Result<()> {
-            let flags = EventFlag::EV_CLEAR | EventFlag::EV_RECEIPT | EventFlag::EV_ADD;
-            let udata = key as _;
-            let changelist = [
-                KEvent::new(fd as _, EventFilter::EVFILT_WRITE, flags, FFLAGS, 0, udata),
-                KEvent::new(fd as _, EventFilter::EVFILT_READ, flags, FFLAGS, 0, udata),
-            ];
-            let mut eventlist = changelist;
-            kevent_ts(self.0, &changelist, &mut eventlist, None).map_err(io_err)?;
-            for ev in &eventlist {
-                // See https://github.com/tokio-rs/mio/issues/582
+        fn kevent_change(
+            &self,
+            raw: RawIO,
+            flags: EventFlag,
+            udata: isize,
+            ignore_epipe: bool,
+        ) -> io::Result<()> {
+            let mut change_buffer = [KEvent::new(
+                0,
+                EventFilter::EVFILT_USER,
+                EventFlag::empty(),
+                FFLAGS,
+                0,
+                0,
+            ); 2];
```

Comment:
It's kind of sad here that I have to create a placeholder value by fully specifying all of its fields. But in the other hand messing with `MaybeUninit` doesn't seem worth it here.